theme/models.py

```diff
@@ -607,7 +607,7 @@ def _assign_bucket_name(self):
         '''Assign a bucket name to the user profile
         The bucket name is derived from the user's username
         '''
-        if hasattr(self, 'user') and self.user is not None:
+        if self.user:
```

Comment:
I think it is worth knowing when there is no user on userprofile so I think you could just remove the if statement entirely.